core/common/src/main/java/alluxio/util/ConfigurationUtils.java

```diff
@@ -247,4 +271,160 @@ public static boolean masterHostConfigured() {
   public static String valueAsString(String value) {
     return value == null ? "(no value set)" : value;
   }
+
+  /**
+   * Returns an instance of {@link AlluxioConfiguration} with the defaults and values from
+   * alluxio-site properties.
+   *
+   * @return The set of Alluxio properties loaded from the site-properties file
+   */
+  public static AlluxioProperties defaults() {
+    if (sDefaultProperties == null) {
+      synchronized (DEFAULT_PROPERTIES_LOCK) { // We don't want multiple threads to reload
+        // properties at the same time.
+        // Check if properties are still null so we don't reload a second time.
+        if (sDefaultProperties == null) {
+          reloadProperties();
+        }
+      }
+    }
+    return sDefaultProperties.copy();
+  }
+
+  /**
+   * Reloads site properties from disk.
+   */
+  public static synchronized void reloadProperties() {
```

Comment:
Do we need both `synchronized(this)` and `synchronized(DEFAULT_PROPERTIES_LOCK)` or can we use just one?

Reply:
Well we don't want to reload the properties at the same time because it's a separate public method, but I would still use the `DEFAULT_PROPERTIES_LOCK` because it stops us from reloading the properties unnecessarily. Without the `synchronize(DEFAULT_PROPERTIES_LOCK)` it's possible for another method to attempt to reload the properties before the reference to the `sDefaultProperties` is set

Reply:
Can we synchronize on `DEFAULT_PROPERTIES_LOCK` here instead of synchronizing on `this`?

Reply:
sure